lib/node_modules/@stdlib/repl/test/integration/test.completer_engine.js

```diff
@@ -0,0 +1,569 @@
+		}
+
+		// Current line is cleared and the original line is inserted:
+		t.strictEqual( data[ data.length-2 ], '\x1B[0K', 'returns expected value' );
+		t.strictEqual( data[ data.length-1 ], 'var zzabc = 1;var zzpqr = 2;zz', 'returns expected value' );
+
+		t.end();
+	}
+});
+
+tape( 'a REPL instance supports displaying highlighted argument TAB completions', function test( t ) {
+	var istream;
+	var opts;
+	var r;
+
+	istream = new DebugStream({
+		'name': 'repl-input-stream'
+	});
+	opts = {
+		'input': istream,
+		'settings': defaultSettings(),
+		'tty': {
+			'rows': 100,
+			'columns': 80
+		}
+	};
+	r = repl( opts, onClose );
+
+	// Declare an object with properties having a common prefix:
+	istream.write( 'var a = { "zzzxy": 4, "zzzab": 6 };\n' );
+
+	// Write the common beginning of the property names in order to generate TAB completions:
+	istream.write( 'a.zzz' );
+
+	// Write TAB to display completions:
+	istream.write( '\t' );
+
+	// Close the input stream:
+	istream.end();
+
+	// Close the REPL:
+	r.close();
+
+	function onClose( error, data ) {
+		var actual;
+
+		if ( error ) {
+			t.fail( error.message );
+			return;
+		}
+
+		// Check for settings name completions in the output:
+		actual = extractCompletions( data[ data.length - 3 ] );
+		t.strictEqual( actual.length, 2, 'returns expected value' );
+		t.ok( contains( actual, '\x1B[1mzzz\x1B[0mab' ), 'returns expected value' );
+		t.ok( contains( actual, '\x1B[1mzzz\x1B[0mxy' ), 'returns expected value' );
+
+		// Check if the cursor is returned back to the prompt:
+		t.strictEqual( data[ data.length - 2 ], '\x1B[2A', 'returns expected value' );
+
+		t.end();
+	}
+});
+
+tape( 'a REPL instance supports auto-completing common prefixes when hitting TAB', function test( t ) {
+	var istream;
+	var opts;
+	var r;
+
+	istream = new DebugStream({
+		'name': 'repl-input-stream'
+	});
+	opts = {
+		'input': istream,
+		'settings': defaultSettings(),
+		'tty': {
+			'rows': 100,
+			'columns': 80
+		}
+	};
+	r = repl( opts, onClose );
+
+	// Declare variables with `zzzz` as an exact prefix:
+	istream.write( 'var zzzz = 1;' );
+	istream.write( 'var zzzzabc = 2;' );
+	istream.write( 'var zzzzpqr = 3;' );
+
+	// Partially write the common beginning of the variable names in order to generate a TAB auto-completion:
+	istream.write( 'zz' );
+
+	// Write TAB to trigger auto-completion:
+	istream.write( '\t' );
+
+	// Close the input stream:
+	istream.end();
+
+	// Close the REPL:
+	r.close();
+
+	function onClose( error, data ) {
+		if ( error ) {
+			t.fail( error.message );
+			return;
+		}
+
+		// Check if the completion prefix was cleared:
+		t.strictEqual( data[ data.length-2 ], '\b\b', 'returns expected value' );
+
+		// Check if the final completion was auto-inserted:
+		t.strictEqual( data[ data.length-1 ], 'zzzz', 'returns expected value' );
+
+		t.end();
+	}
+});
+
+tape( 'a REPL instance supports auto-completing common argument prefixes when hitting TAB', function test( t ) {
```

Comment:
Same comment. What does "argument" refer to here?

Reply:
Initially I was testing with `settings` so was testing for function arguments. I will update it to object properties. Basically I am trying to test for cases where the completion isn't the whole word..